### PR TITLE
Avoid start-up race

### PR DESCRIPTION
**Pull request by LukeShu**
The first 11 commits are me proving to myself that there's no way that there's a start-up race where the interceptor isn't ready in time.  They clean up the code to make it easier for me to reason about; but my conclusion is that there isn't/wasn't a race there (because while prior to these commits the initial call to `interceptor.Start()` didn't block, the following call to `interceptor.Update()` did).

The next 4 commits are otherwise unrelated clean-ups that I noticed I can make.

The final 2 commits actually deal with fixing races.

One of them make _sure_ the DNS server socket is listening before we return from `intercept()`; a theoretical race, but one that shouldn't come up in practice.  Binding the socket is pretty fast, and we do it before several other, much slower, operations that we do properly block on.

The 2nd is to use the [`sd_notify(3)`](https://www.freedesktop.org/software/systemd/man/sd_notify.html) interface to notify our caller when we're fully started.

 - I haven't fully verified that the bridge is fully set up before we send the notification
 - I haven't yet tested this on macOS

### CODE REVIEW

**Comment by LukeShu** [comment]
Now tested to work on macOS.  Changes to make that true:
 - A change I'd made had stopped `nat_pf_test.go` from building, but I hadn't noticed because it only builds on macOS.
 - https://git.lukeshu.com/go/libsystemd/commit/?id=17934614d3030eeebda44b88ce3061d26199a438

File: internal/pkg/interceptor/interceptor.go [hunk 1]
```diff
@@ -215,16 +213,16 @@ func (i *Interceptor) update(table rt.Table) {
...
 
 // GetSearchPath retrieves the current search path
 func (i *Interceptor) GetSearchPath() []string {
-	result := make(chan []string, 1)
-	i.work <- func() {
-		result <- i.search
-	}
-	return <-result
+	i.searchLock.RLock()
+	i.searchLock.RUnlock()
```
Comment (ark3) [inline internal/pkg/interceptor/interceptor.go]: defer here?

```suggestion
	defer i.searchLock.RUnlock()
```

Comment (LukeShu) [reply internal/pkg/interceptor/interceptor.go]: Good catch.

Comment (LukeShu) [reply internal/pkg/interceptor/interceptor.go]: I've amended the relevant commit.

**Pull request merged**